src/network/server.rs

```diff
@@ -591,6 +591,51 @@ impl HubService for MyHubService {
         }
     }
 
+    async fn submit_bulk_messages(
+        &self,
+        request: Request<proto::SubmitBulkMessagesRequest>,
+    ) -> Result<Response<proto::SubmitBulkMessagesResponse>, Status> {
+        authenticate_request(&request, &self.allowed_users)?;
+
+        let messages = request.into_inner().messages;
+        let num_messages = messages.len();
+        info!(
+            "Received call to [submit_bulk_messages] RPC with {} messages",
+            num_messages
+        );
+
+        let mut results = Vec::with_capacity(num_messages);
+        for mut msg in messages {
+            // Ensure data_bytes are decoded before processing
+            message_bytes_decode(&mut msg);
+            let message_hash_for_error = msg.hash.clone();
+
+            // The `submit_message_internal` is async, so we await it so messages are processed sequentially.
+            // We don't want to do parallel processing here, as some message may depend on a previous message.
+            let result = self.submit_message_internal(msg, false).await;
```

Comment:
Note that this doesn't actually wait for the messages to be included in the block, it only simulates it. So if messages depend on each other, they will be simulated independently.

Which will probably become an issue when we get to the next step and support merging both messages at once, we want the simulation to include both messages at the same time. So I think we need to change `simulate_message` to take an array and pass it all the way through.

Also, we should remove the `bypass_validation` flag, it was only used for early performance testing and no one is passing in true anymore.